### PR TITLE
Multi merge keys

### PR DESCRIPTION
**Pull request by naterush**
# Description

Implements spec here: https://www.notion.so/trymito/Multi-Column-Merge-c5290f0e12db417cade11032d9fc68ee

A few minor differences from the spec:
1. Minor formatting differences that look better in practice (IMO, feel free to change though idc)
2. The step params are different from what is proposed, but ends up being pretty similar.
3. Some changes to unique in left and unique in right to make them work

My only thought is that things feel a bit cramped, and maybe we want to just space thing sout a bit more... thoughts?

# Testing

Test the new interface with real merges, and also:
1. Less than two sheets
2. Sheets with no columns

Make sure no sheet crashing bugs as well!

# Documentation

We can update the merge documentation.

### CODE REVIEW

File: mitosheet/mitosheet/tests/test_utils.py [hunk 1]
```diff
@@ -73,6 +73,9 @@ def check_dataframes_equal(test_wrapper):
     # Then, construct code that is just the code we expect, except at the end
     # it compares the dataframe to the final dataframe we expect
     def check_final_dataframe(df_name, df):
+        print("CHECKING")
+        print(final_dfs[df_name])
+        print(df)
```
Comment (naterush) [inline mitosheet/mitosheet/tests/test_utils.py:78]: Remove

**Review by naterush — COMMENTED** [review]
Cleanup!

**Comment by naterush** [comment]
NOTE: this branches off of melt, as I needed some utility changes to it that were hard to cherry pick. So careful if merging.

**Comment by aarondr77** [comment]
Sheet crashing bug:
1. Import data
2. Create df via pivot
3. Open merge taskpane
4. select the pivot dataframe as one of the sheets
5. Undo until the pivot dataframe is gone. 

Highly related to https://github.com/mito-ds/monorepo/issues/227 and https://github.com/mito-ds/monorepo/issues/225

Screen Shot 2022-06-16 at 10 39 14 AM

**Comment by aarondr77** [comment]
Do these overlapping scrollbars cause issues on Windows?

Screen Shot 2022-06-16 at 11 10 49 AM

**Comment by aarondr77** [comment]
If you have two dataframes, one regular data frame and one empty pivot table, then the merge taskpane will open, but it defaults to not having any merge keys and when you click `add merge key` it doesnt' do anything. 

In this case, can we disable the button and have an error message tooltip on hover?

**Comment by aarondr77** [comment]
In terms of the UI, I agree that its pretty cluttered looking. Want to take a shot at cleaning it up?

Btw: one thing that would be helpful next time is to take a screenshot of the initial version of the UI before making changes to it. Cause I think the mockups look better than this, but of course after seeing the mockups implemented that might not be the case. But as a result of not seeing them implemented, I don't have much help to provide to your question about it being cluttered that isn't just want to try the mockup version.

**Review by aarondr77 — COMMENTED** [review]
Okay reviewed! Nice work on this. The taskpane code is a lot cleaner than the previous implementation! 

A few things to callout:
1. I found one sheet crashing bug. Its the classic, undo to remove a df.
2. I think the unique in right/left logic is incorrect. My bad for not specyifing this. 
3. I think you're right that the taskpane looks cluttered. I bet the spacing will go a long way. IMO the mockups look way better, but I guess once implemented they didn't. The one nice thing about the mockups IMO is that there are less shapes. The selects form a column and there isn't that big button that breaks up the entire screen. But I trust your eye for whatever you think looks best implemented.

File: mitosheet/src/components/taskpanes/Merge/MergeTaskpane.tsx [hunk 2]
```diff
@@ -46,168 +47,78 @@ export enum MergeType {
...
+    }
 
-        const suggestedKeys = getSuggestedKeysColumnID(sheetDataArray, sheetOneIndex, sheetTwoIndex);
+    const suggestedMergeKeys = getFirstSuggestedMergeKeys(sheetDataArray, sheetIndexOne, sheetIndexTwo);
 
-        // If there are no suggested keys, then we don't have any columns in one datasets, so we bail with undefined params
-        if (suggestedKeys === undefined) {
-            return undefined;
-        }
+    // We default to selecting _all_ columns, if we switched datasets. Otherwise, we keep the selected
```
[the remark this answers is not in the document]
Comment (naterush) [reply mitosheet/src/components/taskpanes/Merge/MergeTaskpane.tsx:86]: Yeppers!

**Comment by naterush** [comment]
Ok, waiting on new mini-spec, and then LGTM if you're happy with spacing, etc!

**Comment by aarondr77** [comment]
I slightly updated the formatting of the taskpane. I think it improves some things and is slightly worse in other places.

**Pull request merged**